src/app/clusters/thermostat-server/thermostat.cpp

```diff
@@ -45,115 +45,182 @@ void emberAfThermostatClusterServerInitCallback(void)
+
+    emberAfReadAttribute(endpoint, ZCL_THERMOSTAT_CLUSTER_ID, ZCL_MIN_COOL_SETPOINT_LIMIT_ATTRIBUTE_ID, CLUSTER_MASK_SERVER,
+                         (uint8_t *) &MinCoolSetpointLimit, sizeof(MinCoolSetpointLimit), NULL);
+
+    emberAfReadAttribute(endpoint, ZCL_THERMOSTAT_CLUSTER_ID, ZCL_MAX_COOL_SETPOINT_LIMIT_ATTRIBUTE_ID, CLUSTER_MASK_SERVER,
+                         (uint8_t *) &MaxCoolSetpointLimit, sizeof(MaxCoolSetpointLimit), NULL);
+
+    // Make sure the user imposed limits are within the manufacture imposed limits
+
+    if (MinCoolSetpointLimit < AbsMinCoolSetpointLimit)
+        MinCoolSetpointLimit = AbsMinCoolSetpointLimit;
+
+    if (MaxCoolSetpointLimit > AbsMaxCoolSetpointLimit)
+        MaxCoolSetpointLimit = AbsMaxCoolSetpointLimit;
+
+    if (CoolingSetpoint < MinCoolSetpointLimit)
+        CoolingSetpoint = MinCoolSetpointLimit;
+
+    if (CoolingSetpoint > MaxCoolSetpointLimit)
+        CoolingSetpoint = MaxCoolSetpointLimit;
+
+    return CoolingSetpoint;
+}
+
+bool emberAfThermostatClusterSetpointRaiseLowerCallback(EndpointId aEndpointId, chip::app::CommandHandler * commandObj,
+                                                        uint8_t mode, int8_t amount)
 {
-    bool result             = false;
-    EndpointId endpoint     = 1;                            // Hard code to 1 for now/
     int32_t HeatingSetpoint = 2000, CoolingSetpoint = 2600; // Set to defaults to be safe
-    EmberAfStatus status = EMBER_ZCL_STATUS_FAILURE;
+    EmberAfStatus status     = EMBER_ZCL_STATUS_FAILURE;
+    EmberAfStatus ReadStatus = EMBER_ZCL_STATUS_FAILURE;
+
     switch (mode)
     {
     case EMBER_ZCL_SETPOINT_ADJUST_MODE_HEAT_AND_COOL_SETPOINTS: {
+
         // In auto mode we will need to change both the heating and cooling setpoints
-        status = emberAfReadAttribute(endpoint, ZCL_THERMOSTAT_CLUSTER_ID, ZCL_OCCUPIED_COOLING_SETPOINT_ATTRIBUTE_ID,
-                                      CLUSTER_MASK_SERVER, (uint8_t *) &CoolingSetpoint, sizeof(CoolingSetpoint), NULL);
-        if (status == EMBER_ZCL_STATUS_SUCCESS)
+        ReadStatus = emberAfReadAttribute(aEndpointId, ZCL_THERMOSTAT_CLUSTER_ID, ZCL_OCCUPIED_COOLING_SETPOINT_ATTRIBUTE_ID,
+                                          CLUSTER_MASK_SERVER, (uint8_t *) &CoolingSetpoint, sizeof(CoolingSetpoint), NULL);
+        if (ReadStatus == EMBER_ZCL_STATUS_SUCCESS)
         {
-            status = emberAfReadAttribute(endpoint, ZCL_THERMOSTAT_CLUSTER_ID, ZCL_OCCUPIED_HEATING_SETPOINT_ATTRIBUTE_ID,
+            CoolingSetpoint = CoolingSetpoint + static_cast<int16_t>(amount * 10);
+            CoolingSetpoint = EnforceCoolingSetpointLimits(CoolingSetpoint, aEndpointId);
+            status = emberAfWriteAttribute(aEndpointId, ZCL_THERMOSTAT_CLUSTER_ID, ZCL_OCCUPIED_COOLING_SETPOINT_ATTRIBUTE_ID,
+                                           CLUSTER_MASK_SERVER, (uint8_t *) &CoolingSetpoint, ZCL_INT16S_ATTRIBUTE_TYPE);
```

Comment:
So let's say this fails, but the write for heating succeeds.  Should this overall function succeed or fail?  The old code used to fail if either write failed, but this new code ignores the status from the first write if the second write happens...